### PR TITLE
[TE] merge time series snapshot when merging anomalies

### PR DESCRIPTION
**Pull request by vincentchenjl**
This PR is to provide quick fix for merging time series snapshot when merging anomalies.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/ThirdEyeUtilsTest.java [hunk 1]
```diff
@@ -176,21 +179,51 @@ public void testExceptionToStringLimited() {
...
     ThirdEyeUtils.mergeAnomalyProperties(parentProperties, childProperties);
     Assert.assertEquals(parentProperties.get("p1"), "value1");
     Assert.assertEquals(parentProperties.get("p2"), "value2");
     Assert.assertEquals(parentProperties.get("c1"), "value4");
     Assert.assertEquals(parentProperties.get("detectorComponentName"), "rule1,rule2");
+    AnomalyTimelinesView merged = AnomalyTimelinesView.fromJsonString(parentProperties.get("anomalyTimelinesView"));
+    Assert.assertEquals(merged.getTimeBuckets().size(), 11);
+    Assert.assertTrue(merged.getCurrentValues().get(0) - parentVal < 0.00001);
+    Assert.assertTrue(merged.getCurrentValues().get(1) - parentVal < 0.00001);
+    Assert.assertTrue(merged.getCurrentValues().get(10) - childVal < 0.00001);
```
Comment (Ruoyingw) [inline thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/ThirdEyeUtilsTest.java:211]: Probably better to compare with absolute difference: `abs(actual - expected) < 0.00001`. Or better:
we can use `Assert.assertEquals(actual, expected, delta)`

File: thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/ThirdEyeUtilsTest.java [hunk 2]
```diff
@@ -176,21 +179,51 @@ public void testExceptionToStringLimited() {
...
+    parentProperties.put("anomalyTimelinesView",
+        generateAnomalyTimelineView(now, bucketSize, 10, parentVal).toJsonString());
+
 
     Map<String, String> childProperties = new HashMap<>();
     childProperties.put("p1", "value3");
     childProperties.put("c1", "value4");
     childProperties.put("detectorComponentName", "rule2");
+    childProperties.put("anomalyTimelinesView",
+        generateAnomalyTimelineView(now + bucketSize, bucketSize, 10,  childVal).toJsonString());
```
Comment (Ruoyingw) [inline thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/ThirdEyeUtilsTest.java:200]: Can we add another unit test case where the child anomaly has a gap with parent? say 

`childProperties.put("anomalyTimelinesView", generateAnomalyTimelineView(now + bucketSize * 20, bucketSize, 10,  childVal).toJsonString());`

Comment (vincentchenjl) [reply thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/ThirdEyeUtilsTest.java]: Added it as a new test.

File: thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/ThirdEyeUtilsTest.java (hunk 1 above)
Comment (vincentchenjl) [reply thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/util/ThirdEyeUtilsTest.java]: Good point.

**Pull request merged**